### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by mkrout**

### CODE REVIEW

File: services/src/main/java/org/exoplatform/gamification/connectors/github/rest/GithubWebHook.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        try {
+            Map<String, String> gam = new HashMap<>();
+            gam.put("ruleTitle", ruleTitle);
+            gam.put("senderId", senderId);
+            gam.put("receiverId", receiverId);
+            gam.put("object", object);
+            listenerService.broadcast("exo.gamification.generic.action", gam, "");
+            LOG.info("Github action {} gamified for user {} to {}", ruleTitle, senderId, receiverId);
+        } catch (Exception e) {
+            LOG.error("Cannot broadcast gamification event",e);
```
Comment (vsellier) [inline services/src/main/java/org/exoplatform/gamification/connectors/github/rest/GithubWebHook.java]: Please add some info of the event on the log to help the diagnosis

File: services/src/main/resources/db/changelog/github-connector.db.changelog-1.0.0.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        <createTable tableName="GITHUB_ACCOUNT">
+
+            <column name="ID" type="BIGINT" autoIncrement="${autoIncrement}" startWith="1">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="PK_GITHUB_ACCOUNT"/>
+            </column>
+
+            <column name="GITHUB_ID" type="NVARCHAR(50)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="USER_NAME" type="NVARCHAR(50)">
```
Comment (vsellier) [inline services/src/main/resources/db/changelog/github-connector.db.changelog-1.0.0.xml]: Should be 200 ?

File: services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import javax.persistence.*;
+
+import lombok.Data;
+
+import java.io.Serializable;
+
+@Entity(name = "GitHubAccountEntity")
+@ExoEntity
+@Table(name = "GITHUB_ACCOUNT")
```
Comment (vsellier) [inline services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java]: Shouldn't it be prefixed by something?
If I'm not wrong the usage is to pluralize this kind of table names (like ``SOC_IDENTITIES`` ...)

File: services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        ),
+        @NamedQuery(
+                name = "GitHubAccountEntity.getAccountByUserName",
+                query = "SELECT account FROM GitHubAccountEntity account where account.userName = :userName "
+        )
+
+                })
+public class GitHubAccountEntity implements Serializable {
+
+
```
Comment (vsellier) [inline services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java]:
```suggestion
```

File: services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+    @Override
+    public void headerSectionUpdated(ProfileLifeCycleEvent event) {
+
+    }
+
+    @Override
+    public void createProfile(ProfileLifeCycleEvent event) {
+
+
```
Comment (vsellier) [inline services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java]:
```suggestion
```

File: services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+
+    @Override
+    public void aboutMeUpdated(ProfileLifeCycleEvent event) {
+
+
```
Comment (vsellier) [inline services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java]:
```suggestion
```

File: services/src/main/resources/conf/portal/storage-configuration.xml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+<configuration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+               xsi:schemaLocation="http://www.exoplatform.org/xml/ns/kernel_1_2.xsd http://www.exoplatform.org/xml/ns/kernel_1_2.xsd"
+               xmlns="http://www.exoplatform.org/xml/ns/kernel_1_2.xsd">
+
+    <!-- Github ID storage (JPA) -->
+    <component>
+        <type>org.exoplatform.gamification.connectors.github.dao.GitHubAccountDAO</type>
+    </component>
+
+    <!-- Gamification storage layour (Liquebase initializer) -->
```
Comment (vsellier) [inline services/src/main/resources/conf/portal/storage-configuration.xml]:
```suggestion
    <!-- Gamification Connector storage layout (Liquibase initializer) -->
```

File: services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java (hunk 3 above)
Comment (mkrout) [reply services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java]: You're right, and I already added "GITHUB" as a prefix for this addon tables

File: packaging/pom.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    <version>1.0.x-SNAPSHOT</version>
+  </parent>
+  <artifactId>gamification-connectors-packaging</artifactId>
+  <packaging>pom</packaging>
+  <name>eXo Add-on:: eXo Gamification Connectors - Packaging</name>
+  <dependencies>
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>gamification-connectors-war</artifactId>
+      <version>1.0.x-SNAPSHOT</version>
```
Comment (boubaker) [inline packaging/pom.xml]: You should add artifacts of project in parent POM.xml in `dependency-management` tag and avoid to declare here the version. In addition, you can use ${project.version}

File: packaging/src/main/assemblies/packaging.xml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    </dependencySet>
+    <!-- Libraries -->
+    <dependencySet>
+      <useProjectArtifact>false</useProjectArtifact>
+      <outputDirectory>lib</outputDirectory>
+      <includes>
+        <include>${project.groupId}:gamification-connectors-*:jar</include>
+      </includes>
+      <scope>provided</scope>
+      <outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}.${artifact.extension}
```
Comment (boubaker) [inline packaging/src/main/assemblies/packaging.xml]: Please avoid adding version in artifact to ensure that no conflict can be made when developing

File: services/pom.xml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <type>jar</type>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.exoplatform.commons</groupId>
+            <artifactId>commons-webui-ext</artifactId>
```
Comment (boubaker) [inline services/pom.xml]: why using a dependency to webui ?

File: services/pom.xml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.jaxrs</groupId>
+            <artifactId>jackson-jaxrs-base</artifactId>
+            <version>2.4.2</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.exoplatform.social</groupId>
+            <artifactId>social-component-webui</artifactId>
```
Comment (boubaker) [inline services/pom.xml]: why using a dependency to webui ?

File: services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+                        entity.setUserName(event.getUsername());
+                        entity.setGitHubId(gitHubId);
+                        gitHubAccountDAO.create(entity);
+                    } else{
+                        entity.setGitHubId(gitHubId);
+                        gitHubAccountDAO.update(entity);
+                    }
+
+                }else{
+                    LOG.warn("The provided Github ID {} is already used by {}",gitHubId,existingEntity.getUserName());
```
[the remark this answers is not in the document]
Comment (mkrout) [reply services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java]: Nothing to do it means that the Github Id is used and the database will not be updated. maybe in a coming version, we will be able to handle this case, so also the user will be warned in the UI

File: services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java (hunk 3 above)
Comment (boubaker) [reply services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java]: IMO, it should be GAM_GITHUB_ACCOUNTS

File: services/src/main/java/org/exoplatform/gamification/connectors/github/GithubIMSetting.java [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+     * @param profileSettings the profile settings Service
+     * @throws ConfigurationException the configuration exception
+     */
+    public GithubIMSetting(UserProfileSettingsService profileSettings) throws ConfigurationException {
+        this.profileSettings=profileSettings;
+    }
+
+    @Override
+    public void start() {
+        if (profileSettings != null) {
```
[the remark this answers is not in the document]
Comment (mkrout) [reply services/src/main/java/org/exoplatform/gamification/connectors/github/GithubIMSetting.java]: Ok

File: pom.xml [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+    <module>services</module>
+  </modules>
+  <scm>
+    <connection>scm:git:git://github.com/exo-addons/gamification-connectors.git</connection>
+    <developerConnection>scm:git:git@github.com:exo-addons/gamification-connectors.git</developerConnection>
+    <tag>HEAD</tag>
+    <url>https://github.com/exo-addons/gamification-connectors</url>
+  </scm>
+  <properties>
+    <org.exoplatform.platform.version>5.2.0-RC03</org.exoplatform.platform.version>
```
[the remark this answers is not in the document]
Comment (mkrout) [reply pom.xml]: Right

File: packaging/pom.xml (hunk 8 above)
Comment (mkrout) [reply packaging/pom.xml]: Ok

File: services/pom.xml (hunk 10 above)
Comment (mkrout) [reply services/pom.xml]: Wrong dependency

File: services/pom.xml (hunk 11 above)
Comment (mkrout) [reply services/pom.xml]: Wrong dependency

File: packaging/src/main/assemblies/packaging.xml (hunk 9 above)
Comment (mkrout) [reply packaging/src/main/assemblies/packaging.xml]: Ok

File: services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java (hunk 3 above)
Comment (mkrout) [reply services/src/main/java/org/exoplatform/gamification/connectors/github/entity/GitHubAccountEntity.java]: Ok boub, I'm ok with "GAM_GITHUB_ACCOUNTS" also

File: pom.xml [hunk 15]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (C) 2009 eXo Platform SAS.
```
Comment (boubaker) [inline pom.xml]: `Copyright (C) 2019` instead of 2009

File: services/pom.xml [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+        <dependency>
+            <groupId>org.exoplatform.social</groupId>
+            <artifactId>social-component-core</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.16.8</version>
```
Comment (boubaker) [inline services/pom.xml]: the version of lombock should be declared in parent

File: pom.xml (hunk 15 above)
Comment (mkrout) [reply pom.xml]: Ok

File: services/pom.xml (hunk 16 above)
Comment (mkrout) [reply services/pom.xml]: Ok

File: services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (vsellier) [reply services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java]: ok so it should be a ``INFO`` level.

[the remark this answers is not in the document]
Comment (mkrout) [reply services/src/main/java/org/exoplatform/gamification/connectors/github/listener/GithubProfileListener.java]: ok

**Pull request merged**